Dokan/Dokan/Scenes/ProductDetails/ReviewsView.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ReviewsView.swift
+//  Dokan
+//
+//  Created by Clara Mounir Adly on 16/07/2022.
+//
+
+import Foundation
+import UIKit
+
+class ReviewsView: UIView, UITableViewDelegate, UITableViewDataSource {
+    
+    var uiMODEL: ReviewsUiModel?
+   // MARK: outlets
+    @IBOutlet weak var contentView: UIView!
+    @IBOutlet weak var reviews: UILabel!
+    @IBOutlet weak var rating: UILabel!
+    @IBOutlet weak var customerReview: UITableView!
+    @IBOutlet weak var seeAllReviewsButton: UIButton!
+    
+    // MARK: functions
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        uiMODEL!.reviews.count
+    }
+    
+   // func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+ //       guard let cell = tableView.dequeueReusableCell(withIdentifier: " ") else { return // } 
+      
+    //    return cell
+  //  }
+    
+    override init(frame: CGRect) {
+          super.init(frame: frame)
+          commonInit()
+        
+        customerReview.dataSource = self
+        customerReview.delegate = self
+      }
+      required init?(coder aDecoder: NSCoder) {
+          super.init(coder: aDecoder)
+          commonInit()
+      }
+      func commonInit() {
+          Bundle.main.loadNibNamed("ReviewsView", owner: self, options: nil)
+          contentView.fixInView(self)
+      }
+  }
```

Comment:
- Never keep any commented code, gIt should take care of history.
- Parameters and methods order is convention, please stick to it. (Outlets - Parameters - Methods - Init)
- Data hiding is great for you code, please make use of it. 
- Decorator design pattern is great when you attempt to use protocols. It's better to add them in extension. 

```suggestion
class ReviewsView: UIView {
    
    // MARK: Outlets
    
    @IBOutlet private weak var contentView: UIView!
    @IBOutlet private weak var reviewsLabel: UILabel!
    @IBOutlet private weak var ratingLabel: UILabel!
    @IBOutlet private weak var tableView: UITableView!
    @IBOutlet private weak var seeAllReviewsButton: UIButton!
    
    // MARK: Properties
    
    var viewModel: ViewModel?

    // MARK: Init
    
    override init(frame: CGRect) {
          super.init(frame: frame)
          commitInit()
      }
      
      required init?(coder aDecoder: NSCoder) {
          super.init(coder: aDecoder)
          commitInit()
      }
      
      private func commitInit() {
          // TODO: - Use `className` extension method on UIView instead of hard coded values. 
          Bundle.main.loadNibNamed("ReviewsView", owner: self, options: nil)
          contentView.fixInView(self)
          configureTableView()
      }
}

// MARK: - Configure View
//
private extension ReviewsView { 

    func configureTableView() { 
          tableView.dataSource = self
          tableView.delegate = self
    }
}

// MARK: - UITableViewDelegate
//
extension ReviewsView: UITableViewDelegate { 
}

// MARK: - UITableViewDataSource
//
extension ReviewsView: UITableViewDataSource { 

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        viewModel?.reviews.count ?? .zero
    }
}

// MARK: - Nested Types
//
extension ReviewsView { 

    struct ViewModel { 
        let reviewsCount: Int
        let rate: Double
        let reviews: [Review]
    }
}
```